features/gov_uk.feature

```diff
@@ -6,25 +6,91 @@ Feature: Core GOV.UK behaviour
     Given I am testing through the full stack
     And I force a varnish cache miss
 
+  # TODO: REMOVE this test as it does not meet the eligibility
+  # criteria in docs/writing-tests.md.
+  #
+  # - Covers site-wide config: N (edge case)
+  # - Targets data transfer: N (not applicable)
+  # - Second critical check: N (not applicable)
+  #
+  # The low impact and likelihood of this breaking compared to
+  # any other CDN feature means the risk of removal is low.
+  #
+  # No evidence this is a vital feature [^1].
+  #
+  # [^1]: https://github.com/alphagov/smokey/commit/436cdce8460b22baf654e1f3342ae36846bb6567
+  #
   Scenario: Check paths with a trailing slash are redirected
     When I visit "/browse/benefits/" without following redirects
     Then I should get a 301 status code
     And I should be at a location path of "/browse/benefits"
 
+  # TODO: REMOVE this test as it does not meet the eligibility
+  # criteria in docs/writing-tests.md.
+  #
+  # - Covers site-wide config: N (edge case)
+  # - Targets data transfer: N (not applicable)
+  # - Second critical check: N (not applicable)
+  #
+  # The low impact and likelihood of this breaking compared to
+  # any other CDN feature means the risk of removal is low.
+  #
+  # No evidence this is a vital feature [^1].
+  #
```

Comment:
I think things like email clients can be a bit naive about auto-linking URLs and sometimes mistakenly include full stops. I think there's a risk of people thinking this isn't vital and it breaking and departmental coms leading to users not seeing pages. Inclined to keep this.

Reply:
Thanks for challenging this - I struggled with these so appreciate the input.

It sounds like this is different to the other scenarios, which are focussed on human error: you're saying this could actually manifest through software, which does add weight to it being useful.

My worry with this is that it sets a precedent for having other edge-case tests: while a full stop may be more of an issue, I suspect it still accounts for a small minority of our daily requests.

Here are some options:

1. I can add a commit to keep this test until we have data to justify it's not vital e.g. if it only applies to .00000001% of our daily requests, I think that is a strong justification that this test has very low value.

2. We can continue with the removal. There's always a risk with removing a test but we can still agree it's worth the risk, even with this new knowledge. I still lean towards removal, but it's ultimately subjective.

With that in mind, which option do you think I should go with?

Reply:
Fair enough.

@kevindew @richardTowers could one of you give us a steer on whether this is an acceptable risk?

It's just one test, so I don't feel strongly - we can keep it if you're not sure.

Reply:
Sure, I've just had a look into it.

I can see from [Kibana](https://kibana.logit.io/s/2dd89c13-a0ed-4743-9440-825e2e52329e/goto/ff16963394a0b4f914b23737de857e48) that we've had ~30,000 of requests that get this redirect compared to ~177,000,000 requests - so perhaps 0.02% of requests that aren't handled at CDN (which still seems sufficiently high that I fear I've made a mistake).

I think it's an acceptable risk. Ideally this would be tested with the VCL file that configures this: https://github.com/alphagov/govuk-puppet/blob/32c1bbbb10067078c1406170666a135b4a10aaea/modules/varnish/templates/default.vcl.erb 

So I'd think it'd be valuable here if we decided that this is the suite to test varnish and we had a varnish.feature which tests each thing added. As it's currently a hodge-podge I think a better precedent would be to remove them, rather than adding a bunch of missing varnish tests (for consistency). It doesn't seem ideal to have a situation where we selectively test bits of VCL.

Reply:
> so perhaps 0.02% of requests that aren't handled at CDN

Ooh thanks for checking. I'm surprised I have access to Production Kibana 🤔.

> which still seems sufficiently high that I fear I've made a mistake

Yes, I think it's even lower: looking through the logs I can see we're also catching query params and invalid URLs that happen to end in a `.`. We're likely seeing a 301 for the bad URLs due to `http => https`.

I'm not sure how to do better, but 0.02% is low. The scenario of a bad link in an email is still possible, but we know most of our traffic doesn't come from these - at least, not from the emails we send ourselves.

I'm going to resolve this one as we've agreed it's an acceptable risk.